core/include/io.h

```diff
@@ -273,4 +273,22 @@ static inline void io_clrsetbits8(vaddr_t addr, uint8_t clear_mask,
 	io_write8(addr, (io_read8(addr) & ~clear_mask) | set_mask);
 }
 
+#define readl_poll_timeout(addr, val, cond, delay_us, timeout_us) \
```

Comment:
An inline description comment is welcome.
Perfer use uppercase characters for macros (`READL_POLL_TIMEOUT()`)
As this relies on `io_read32()`, I think macro name `IO_READ32_POLL_TIMEOUT()` would be better.